src/Nethermind/Nethermind.Core/Caching/StaticPoolcs.cs

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace Nethermind.Core.Caching;
+
+public static class StaticPool<T> where T : class, new()
```

Comment:
It does not guarantee it will return clean items, seems a bit risky. It can additionally reset items